includes/class-wc-stripe-helper.php

```diff
@@ -752,4 +752,85 @@ public static function add_payment_intent_to_order( $payment_intent_id, $order )
+	 * payment method ID is provided. If ID is neither a source or a payment method ID then nothing
+	 * is added.
+	 *
+	 * @param string $payment_method_id  The payment method ID that should be added to the request array.
+	 * @param array $request             The request representing the arguments that will be sent in the request.
+	 *
+	 * @return array  The updated request array.
+	 */
+	public static function add_payment_method_to_request_array( string $payment_method_id, array $request ): array {
+		if ( 0 === strpos( $payment_method_id, 'src_' ) ) {
+			$request['source'] = $payment_method_id;
+		} elseif ( 0 === strpos( $payment_method_id, 'pm_' ) ) {
+			$request['payment_method'] = $payment_method_id;
+		}
+
+		return $request;
+	}
+
+	/**
+	 * Evaluates whether the object passed to this function is a Stripe Payment Method.
+	 *
+	 * @param stdClass $object  The object that should be evaluated.
+	 * @return bool             Returns true if the object is a Payment Method; false otherwise.
+	 */
+	public static function is_payment_method_object( stdClass $payment_method ): bool {
+		return isset( $payment_method->object ) && 'payment_method' === $payment_method->object;
+	}
+
+	/**
+	 * Evaluates whether a given Stripe Source (or Stripe Payment Method) is reusable.
+	 * Payment Methods are always reusable; Sources are only reusable when the appropriate
+	 * usage metadata is provided.
+	 *
+	 * @param stdClass $payment_method  The source or payment method to be evaluated.
+
+	 * @return bool  Returns true if the source is reusable; false otherwise.
+	 */
+	public static function is_reusable_payment_method( stdClass $payment_method ): bool {
+		return self::is_payment_method_object( $payment_method ) || ( isset( $payment_method->usage ) && 'reusable' === $payment_method->usage );
+	}
+
+	/**
+	 * Returns true if the provided payment method is a card, false otherwise.
+	 *
+	 * @param stdClass $payment_method  The provided payment method object. Can be a Source or a Payment Method.
+	 *
+	 * @return bool  True if payment method is a card, false otherwise.
+	 */
+	public static function is_card_payment_method( stdClass $payment_method ): bool {
+		if ( ! isset( $payment_method->object ) || ! isset( $payment_method->type ) ) {
+			return false;
+		}
+
+		if ( 'payment_method' !== $payment_method->object && 'source' !== $payment_method->object ) {
+			return false;
+		}
+
+		return 'card' === $payment_method->type;
+	}
+
+	/**
+	 * Returns a source or payment method from a given intent object.
+	 *
+	 * @param stdClass|object $intent  The intent that contains the payment method.
+	 *
+	 * @return stdClass|string|null  The payment method if found, null otherwise.
+	 */
+	public static function get_payment_method_from_intent( $intent ) {
+		if ( isset( $intent->source ) ) {
+			return $intent->source;
+		}
+
+		if ( isset( $intent->payment_method ) ) {
+			return $intent->payment_method;
+		}
```

Comment:
Maybe we should check for `! empty()` to be safer?

Reply:
Nice catch, fixed in 8e279d40d53fb6126aed22e2336e9dff444cb005!